### ISSUE TITLE
When asserting maps, the way errors are reported is wrong

### ISSUE THREAD

**Issue by Zordid** [kotest/kotest#2206]
**Which version of Kotest are you using**
4.4.3

When I assert that "SOMETHING *shouldBe* EXPECTED", I expect deviations from that to sound like this:

"Expected SOMETHING to be EXPECTED"

But strangely, when you do this for maps, it is the other way around - the classic confusion with actual and expected.

If you run this:
```
mapOf("field" to "actual") shouldBe mapOf("field" to "expected")
```

The output says:
Expected
{
  "field" = "expected"
}
to be equal to
{
  "field" = "actual"
}
Values differed at keys field

In other words: it claims that it expects the expectation to be like the actual - which is the wrong way around. :-(

In real life, I do not change my expectation, but rather my actual value.

**Comment by Kantis** [comment]
I started to look into this and came to these conclusions:
* I believe this comes from the way Junit5 handles `AssertionFailedError`s (see [here](https://github.com/junit-team/junit5/blob/43638eb6a870e0d6c49224053dfeb39dcf0ef33f/junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionUtils.java#L122)) 
* I believe the reason for using `AssertionFailedError` is to get IntelliJ to show the `<Click to see difference>` action in the logs.
* Hence I believe it's a hard to change this in Kotest. 

The map comparison should probably pass the actual and expected to the `AssertionFailedError` so they show up for diffing, however. I started a PR for that here: https://github.com/kotest/kotest/pull/2311
